libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/PemUtils.java

```diff
@@ -213,6 +213,13 @@ private static PrivateKey parsePKCS8(BufferedReader bReader) throws IOException,
         return keyFactory.generatePrivate(new PKCS8EncodedKeySpec(keyBytes));
     }
 
+    public static PrivateKey parsePKCS8PemString(String pem) throws IOException, GeneralSecurityException{
+        byte[] keyBytes = Base64.getDecoder().decode(pem);
+        String keyAlgo = getKeyAlgorithmIdentifier(keyBytes);
+        KeyFactory keyFactory = KeyFactory.getInstance(keyAlgo);
+        return keyFactory.generatePrivate(new PKCS8EncodedKeySpec(keyBytes));
+    }
```

Comment:
This looks like it's duplicated code from the bottom of `parsePKCS8` can we delegate that method here instead of copy-paste?

Reply:
yes we can, that was sloppy